google-cloud-pubsub/lib/google/cloud/pubsub/service.rb

```diff
@@ -173,8 +173,10 @@ def delete_topic topic_name
         # Raises GRPC status code 5 if the topic does not exist.
         # The messages parameter is an array of arrays.
         # The first element is the data, second is attributes hash.
-        def publish topic, messages
-          publisher.publish topic: topic_path(topic), messages: messages
+        def publish topic, messages, compress: false
+          request = { topic: topic_path(topic), messages: messages }
+          compress_options = ::Gapic::CallOptions.new metadata: { "grpc-internal-encoding-request": "gzip" }
+          compress ? (publisher.publish request, compress_options) : (publisher.publish request)
```

Comment:
Since the `options` is an optional parameter, can we remove the ternary operator?

Reply:
Only when compression is enabled we will have to pass the compress_options. The check has to be somewhere hence. I feel this is better than putting at compress_options creation as existing mocks in tests need not be updated.